pkg/controller/parser/docker_config_parser_test.go

```diff
@@ -0,0 +1,57 @@
+package parser
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+)
+
+func TestNewDockerConfig(t *testing.T) {
+	t.Run("NewDockerConfig", func(t *testing.T) {
+		auths := make(map[string]struct {
```

Comment:
In another class, a type `auth` is used over `struct {Username string, Password string}`.